backend/controller/dal/fsm.go

```diff
@@ -57,11 +58,15 @@ func (d *DAL) StartFSMTransition(ctx context.Context, fsm schema.RefKey, executi
 		}
 		return fmt.Errorf("failed to start FSM transition: %w", err)
 	}
+
+	observability.RecordFsmTransitionBegin(ctx, fsm)
```

Comment:
nit: go convention is to fully capitalize all initialisms, so this would be `RecordFSMTransitionBegin`. Same elsewhere